src/main/java/ladysnake/illuminations/common/entities/FireflyEntity.java

```diff
@@ -30,47 +33,62 @@
     protected boolean isAttractedByLight;
     protected int nextAlphaGoal;
     protected int offset;
+    
+    private AttributeContainer attributes;
 
     // Constructors
-    public FireflyEntity(EntityType entityType, World world) {
+    public <T extends LightOrbEntity> FireflyEntity(EntityType<T> entityType, World world) {
         super(entityType, world);
 
         this.scaleModifier = 0.1F + new Random().nextFloat() * 0.15F;
         this.colorModifier = 0.25F + new Random().nextFloat() * 0.75F;
-        this.alpha = 255;
-
-        this.canDespawn = true;
-        this.isAttractedByLight = true;
-
         this.offset = this.getRandom().nextInt(20);
-
-        this.getAttributeInstance(EntityAttributes.MAX_HEALTH).setBaseValue(1.0D);
+        this.isAttractedByLight = true;
+        this.canDespawn = true;
+        this.alpha = 255;
     }
 
     public FireflyEntity(World world, double x, double y, double z) {
         this(IlluminationsEntities.FIREFLY, world);
         this.updatePosition(x, y, z);
     }
-
+    
+    private DefaultAttributeContainer initAttributes() {
+    	return MobEntity.createMobAttributes()
+						.add(EntityAttributes.GENERIC_MAX_HEALTH, 1.0)
+						.add(EntityAttributes.GENERIC_FLYING_SPEED, 0.6)
+						.add(EntityAttributes.GENERIC_MOVEMENT_SPEED, 0.25)
+						.add(EntityAttributes.GENERIC_FOLLOW_RANGE, 48.0)
+						.build();
+    }
+    
+    @Override
+    public AttributeContainer getAttributes() {
+    	if (attributes == null) {
+    		this.attributes = new AttributeContainer(this.initAttributes());
+    	}
+    	return this.attributes;
+    }
```

Comment:
This should be replaced with a call to `FabricDefaultAttributeRegistry#register`